src/CachePlugin.php

```diff
@@ -183,11 +236,12 @@ protected function isCacheable(ResponseInterface $response)
         if (!in_array($response->getStatusCode(), [200, 203, 300, 301, 302, 404, 410])) {
             return false;
         }
-        if (!$this->config['respect_cache_headers']) {
-            return true;
-        }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
```

Comment:
Removing this will also break BC, right?

Can we do something smart in the constructor? If `respect_cache_headers` is true then populate `respect_response_cache_directives` with "no-store", "private".